src/opening-period/time-span/TimeSpan.tsx

```diff
@@ -35,55 +41,70 @@ export default function TimeSpan({
-          <label htmlFor={`timeSpans[${index}].startTime`}>Kellonaika *</label>
+      <div className="form-control">
+        <div className="start-and-end-time-container">
+          <div className="start-time-container">
+            <label
+              htmlFor={`timeSpans[${index}].startTime`}
+              className="start-time-container-label">
+              Kellonaika *
+            </label>
+            <TimeInput
+              ariaLabel="Aukiolon alkukellonaika"
+              dataTest={`time-span-start-time-${index}`}
+              register={register}
+              defaultValue={`${item.startTime}`}
+              required
+              id={`timeSpans[${index}].startTime`}
+              name={`timeSpans[${index}].startTime`}
+              placeholder="--.--"
+            />
+          </div>
+          <div className="dash-between-start-and-end-time-container">
+            <p>—</p>
+          </div>
           <TimeInput
-            ariaLabel="Aukiolon alkukellonaika"
-            dataTest={`time-span-start-time-${index}`}
+            ariaLabel="Aukiolon loppukellonaika"
+            dataTest={`time-span-end-time-${index}`}
             register={register}
+            defaultValue={`${item.endTime}`}
             required
-            value=""
-            id={`timeSpans[${index}].startTime`}
-            name={`timeSpans[${index}].startTime`}
+            id={`timeSpans[${index}].endTime`}
+            name={`timeSpans[${index}].endTime`}
             placeholder="--.--"
           />
         </div>
-        <div className="dash-between-start-and-end-time-container">
-          <p>—</p>
-        </div>
-        <TimeInput
-          ariaLabel="Aukiolon loppukellonaika"
-          dataTest={`time-span-end-time-${index}`}
-          register={register}
-          required
-          value=""
-          id={`timeSpans[${index}].endTime`}
-          name={`timeSpans[${index}].endTime`}
-          placeholder="--.--"
-        />
       </div>
-      <div className="time-span-state-container">
-        <Select
-          id={`time-span-state-id-${index}`}
-          onChange={(selection: { label: string; value: string }): void => {
-            setValue(`timeSpans[${index}].resourceState`, selection.value);
-          }}
-          className="time-span-state"
-          options={resourceStateOptions}
-          label="Tyyppi"
-          placeholder="Valitse tyyppi"
-        />
+      <div className="form-control">
+        <div className="time-span-state-container">
+          <Select
+            id={`time-span-state-id-${index}`}
+            onChange={(selection: { label: string; value: string }): void => {
+              setValue(`timeSpans[${index}].resourceState`, selection.value);
+            }}
+            className="time-span-state"
+            defaultValue={resourceStateOptions.find(
+              ({ value }) =>
+                value === ((item.resourceState as unknown) as string)
```

Comment:
Why do we first cast it to unknown and then to string?

Reply:
This the way to supress enimmäkseen to a string. We do this also elsewhere